docs/proposals/env-vars.md

```diff
@@ -0,0 +1,563 @@
+<!--
+Copyright The Shipwright Contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+---
+title: Expressing Environment Variables in Shipwright API
+authors:
+  - "@gabemontero"
+
+reviewers:
+  - "@sbose78"
+  - "@SaschaSchwarze0"
+  - "@otaviof"
+<!-- 
+Note from Gabe: I went light (i.e. just Shoubhik since he asked for this EP) on reviewers as a first pass,
+only because it implied a mandate / expectation.  I've chosen to update this list and folks actually review.
+-->
+
+approvers:
+  - "@sbose78"
+<!-- 
+Note from Gabe: I went light (i.e. just Shoubhik since he asked for this EP) on approvers as a first pass,
+only because it implied a mandate / expectation.  I've chosen to update this list and folks with approve permissions actually review,
+where I'll certainly make sure we get multi company sign off.
+-->
+
+creation-date: 2021-04-09
+
+last-updated: 2021-04-27
+
+status: implementable    
+
+<!-- status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced -->
+see-also:
+  - [Parameterize Build Strategies](https://github.com/shipwright-io/build/pull/697/files)
+---
+
+# Expressing Environment Variables in Shipwright API
+
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Open Questions [optional]
+
+> 1. API change
+
+Assuming this gets delivered while Shipwright builds is still v1alpha1, we just have to announce to users
+what new API is provided.
+
+But if the implementation of this is delayed to the point that we are at say v1beta1, then we have to 
```

Comment:
I don't think we can confidently ship a v1beta1 API without this implemented.

Reply:
part of next push